### PR TITLE
Networking Layer

### PR DESCRIPTION
**Pull request by akshitverma4**
Following Files have been created for Networking Layer (Retrofit Setup):

1. [Package: model]
- IfscResponse 

2. [Package: network]
- IfscSearchApi
- RetrofitInstance

3. [Package: util]
- Resource

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -14,14 +14,49 @@ android {
...
+            versionNameSuffix "-debug"
+            buildConfigField "String", "TYPE", '"I AM A DEBUG NINJA"'
+            resValue "string", "my_name", "Anupam Debugger"
+        }
+        beta{
+            signingConfig signingConfigs.beta
+            applicationIdSuffix ".beta"
+            versionNameSuffix "-beta"
+            buildConfigField "String", "TYPE", '"I AM A BETA NINJA"'
+            resValue "string", "my_name", "Anupam Beta"
```
Comment (aliHafizji) [inline app/build.gradle:51]: Why do you need this?

File: app/build.gradle [hunk 2]
```diff
@@ -14,14 +14,49 @@ android {
...
+            applicationIdSuffix ".debug"
+            versionNameSuffix "-debug"
+            buildConfigField "String", "TYPE", '"I AM A DEBUG NINJA"'
+            resValue "string", "my_name", "Anupam Debugger"
+        }
+        beta{
+            signingConfig signingConfigs.beta
+            applicationIdSuffix ".beta"
+            versionNameSuffix "-beta"
+            buildConfigField "String", "TYPE", '"I AM A BETA NINJA"'
```
Comment (aliHafizji) [inline app/build.gradle:50]: Why do you need this?

File: app/build.gradle [hunk 3]
```diff
@@ -14,14 +14,49 @@ android {
...
+            resValue "string", "my_name", "Anupam Beta"
+        }
+    }
+    productFlavors{
+        free{
+            applicationId "com.wednesday.banklocator"
+        }
+        paid{
```
Comment (aliHafizji) [inline app/build.gradle:58]: Why are you using free and paid? You should use beta & prod

File: app/src/main/java/com/wednesday/banklocator/model/IfscResponse.kt [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+data class IfscResponse(
+    val ADDRESS: String,
+    val BANK: String,
+    val BRANCH: String,
+    val CITY: String,
+    val CONTACT: String,
+    val DISTRICT: String,
+    var IFSC: String,
+    val RTGS: Boolean,
+    val STATE: String
```
Comment (aliHafizji) [inline app/src/main/java/com/wednesday/banklocator/model/IfscResponse.kt]: Why is everything capital?

File: app/src/main/java/com/wednesday/banklocator/network/IfscSearchApi.kt [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import retrofit2.Response
+import retrofit2.http.GET
+import retrofit2.http.Path
+
+interface IfscSearchApi {
+
+    @GET("/{ifscCode}")
+    suspend fun searchForIFSC(
+        @Path("ifscCode")
+        Y: String
```
Comment (aliHafizji) [inline app/src/main/java/com/wednesday/banklocator/network/IfscSearchApi.kt]: What is `Y`. Please rename

File: app/src/main/java/com/wednesday/banklocator/network/RetrofitInstance.kt [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    companion object {
+
+        private val retrofit by lazy {
+            val logging = HttpLoggingInterceptor()
+            logging.setLevel(HttpLoggingInterceptor.Level.BODY)
+            val client = OkHttpClient.Builder()
+                .addInterceptor(logging)
+                .build()
+            Retrofit.Builder()
+                .baseUrl("https://ifsc.razorpay.com/")
```
Comment (aliHafizji) [inline app/src/main/java/com/wednesday/banklocator/network/RetrofitInstance.kt]: Please move to a constant variable

**Review by aliHafizji — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/wednesday/banklocator/model/IfscResponse.kt (hunk 4 above)
Comment (akshitverma4) [reply app/src/main/java/com/wednesday/banklocator/model/IfscResponse.kt]: Some of the files including this "data class" are auto-generated using the ".json to class object" plugin.
But I'll change them

File: app/src/main/java/com/wednesday/banklocator/network/IfscSearchApi.kt (hunk 5 above)
Comment (akshitverma4) [reply app/src/main/java/com/wednesday/banklocator/network/IfscSearchApi.kt]: Sure, I'll renane it.

File: app/src/main/java/com/wednesday/banklocator/network/RetrofitInstance.kt (hunk 6 above)
Comment (akshitverma4) [reply app/src/main/java/com/wednesday/banklocator/network/RetrofitInstance.kt]: Got it.

File: app/src/main/java/com/wednesday/banklocator/model/IfscResponse.kt (hunk 4 above)
Comment (akshitverma4) [reply app/src/main/java/com/wednesday/banklocator/model/IfscResponse.kt:12]: I'll change them.

File: app/src/main/java/com/wednesday/banklocator/network/RetrofitInstance.kt [hunk 7]
```diff
@@ -8,6 +8,8 @@ import retrofit2.converter.gson.GsonConverterFactory
 
 class RetrofitInstance {
     companion object {
+        
+        const val baseUrl:String = "https://ifsc.razorpay.com/"
```
Comment (aliHafizji) [inline app/src/main/java/com/wednesday/banklocator/network/RetrofitInstance.kt]: Use type inference. You don't need to declare the type when you're defining it.

Comment (akshitverma4) [reply app/src/main/java/com/wednesday/banklocator/network/RetrofitInstance.kt]: sure thing!

**Review by aliHafizji — APPROVED** [review]

**Pull request merged**